dvc/render/image_converter.py

```diff
@@ -0,0 +1,58 @@
+import base64
+import os
+from typing import TYPE_CHECKING, Dict, List, Optional, Tuple
+
+from dvc.render import REVISION_FIELD, SRC_FIELD
+
+if TYPE_CHECKING:
+    from dvc.types import StrPath
+
+
+class ImageConverter:
+    def __init__(self, plot_properties: Optional[Dict] = None):
+        self.plot_properties = plot_properties or {}
+
+    @staticmethod
+    def _write_image(
+        path: "StrPath",
+        revision: str,
+        filename: str,
+        image_data: bytes,
+    ) -> "StrPath":
+        img_path = os.path.join(
+            path, f"{revision}_{filename.replace(os.sep, '_')}"
+        )
+        with open(img_path, "wb") as fd:
+            fd.write(image_data)
+
+        return img_path
+
+    @staticmethod
+    def _encode_image(
+        image_data: bytes,
+    ) -> str:
+        base64_str = base64.b64encode(image_data).decode()
+        return f"data:image;base64,{base64_str}"
+
+    def convert(
+        self, revision, filename, data: bytes
+    ) -> Tuple[List[Dict], Dict]:
+        """
+        Convert the DVC Plots content to DVC Render datapoints.
+        Return both generated datapoints and updated properties.
+        """
+        path = self.plot_properties.get("out")
+        if path:
```

Comment:
@pared I made `out` optional here. This should still work as before when `out` is provided but now also works without output folder by encoding images with base64.